compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -7576,6 +7576,9 @@ private BType checkMappingField(RecordLiteralNode.RecordField field, BType mappi
                     boolean errored = false;
                     for (BField bField : ((BRecordType) spreadExprType).fields.values()) {
                         BType specFieldType = bField.type;
+                        if (specFieldType.tag ==  TypeTags.NEVER) {
```

Comment:
We expect only one instance of the never type, which is `org.wso2.ballerinalang.compiler.semantics.model.SymbolTable#neverType`.

So we can do

```suggestion
                        if (specFieldType ==  symTable.neverType) {
```

Reply:
Anyway, this will not work when the type is effectively `never`. E.g.,

```ballerina
record {|
    never x;
|}
```

or 

```ballerina
never|never
```

You will have to use `org.wso2.ballerinalang.compiler.semantics.analyzer.Types#isNeverTypeOrStructureTypeWithARequiredNeverMember(org.wso2.ballerinalang.compiler.semantics.model.types.BType)`.

Please add a few tests too.